runtime/src/set_auth_manifest.rs

```diff
@@ -0,0 +1,504 @@
+            )
+            .map_err(|_| CaliptraError::RUNTIME_AUTH_MANIFEST_OWNER_LMS_SIGNATURE_INVALID)?;
+            let pub_key_digest =
+                HashValue::from(auth_manifest_preamble.owner_pub_keys.lms_pub_key.digest);
+            if candidate_key != pub_key_digest {
+                Err(CaliptraError::RUNTIME_AUTH_MANIFEST_OWNER_LMS_SIGNATURE_INVALID)?;
+            } else {
+                caliptra_cfi_lib_git::cfi_assert_eq_6_words(&candidate_key.0, &pub_key_digest.0);
+            }
+        }
+
+        Ok(())
+    }
+
+    fn process_image_metadata_col(
+        cmd_buf: &[u8],
+        auth_manifest_preamble: &AuthManifestPreamble,
+        image_metadata_col: &mut AuthManifestImageMetadataCollection,
+        sha384: &mut Sha384,
+        ecc384: &mut Ecc384,
+        sha256: &mut Sha256,
+        soc_ifc: &SocIfc,
+    ) -> CaliptraResult<()> {
+        if (cmd_buf.len() < size_of::<AuthManifestImageMetadataCollectionHeader>()) {
+            Err(CaliptraError::RUNTIME_AUTH_MANIFEST_IMAGE_METADATA_LIST_INVALID_SIZE)?;
+        }
+
+        let col_size = min(
+            cmd_buf.len(),
+            size_of::<AuthManifestImageMetadataCollection>(),
+        );
+        let buf = cmd_buf
+            .get(..col_size)
+            .ok_or(CaliptraError::RUNTIME_AUTH_MANIFEST_IMAGE_METADATA_LIST_INVALID_SIZE)?;
+
+        image_metadata_col.as_bytes_mut()[..col_size].copy_from_slice(buf);
+
+        if (image_metadata_col.header.entry_count == 0
+            || image_metadata_col.header.entry_count
+                > AUTH_MANIFEST_IMAGE_METADATA_LIST_MAX_COUNT as u32)
+        {
+            Err(CaliptraError::RUNTIME_AUTH_MANIFEST_IMAGE_METADATA_LIST_INVALID_ENTRY_COUNT)?;
+        }
+
+        let digest_metadata_col = Self::sha384_digest(sha384, buf, 0, col_size as u32)?;
+
+        Self::verify_vendor_image_metadata_col(
+            auth_manifest_preamble,
+            &digest_metadata_col,
+            sha384,
+            ecc384,
+            sha256,
+            soc_ifc,
+        )?;
+
+        Self::verify_owner_image_metadata_col(
+            auth_manifest_preamble,
+            &digest_metadata_col,
+            sha384,
+            ecc384,
+            sha256,
+            soc_ifc,
+        )?;
+
+        Ok(())
+    }
+
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    #[inline(never)]
+    pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
+        // Validate cmd length
+        let manifest_size = {
+            let err = CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS;
+            let offset = offset_of!(SetAuthManifestReq, manifest_size);
+            let size = u32::from_le_bytes(
+                cmd_args
+                    .get(offset..offset + 4)
+                    .ok_or(err)?
+                    .try_into()
+                    .map_err(|_| err)?,
+            );
+            usize::try_from(size).map_err(|_| err)?
```

Comment:
Are we worried about the conversion of `u32` to `usize` failing? It seems like `as usize` would do the same thing.

I also don't think the `try_into()` in the above line can fail unless somehow `u32` is not 4 bytes, but `.unwrap()` takes about the same amount of space as `.map_err(|_| err)`.

Reply:
Removed.